src/services/inplayer.account.service.ts

```diff
@@ -120,7 +120,7 @@ export const logout = async () => {
     InPlayer.Notifications.unsubscribe();
     await InPlayer.Account.signOut();
   } catch {
-    throw new Error('Failed to sign out.');
+    console.error('Failed to sign out.');
```

Comment:
Why did you change this? Aren't there cases where upstream code will want to give feedback to viewers when the logout request fails?

Reply:
We don't catch this error and that's the reason it breaks the simulations logins functionality.

Reply:
Can you please elaborate on that? I don't like burying errors like this.

Reply:
By changing this, you are changing not just the simultaneous logout flow, but also the normal logout flow. The user might try to sign out and have it fail, then there's no longer any way for us to give feedback.

If logout always fails in this scenario why are we even calling it? Can we just call `InPlayer.Notifications.unsubscribe`? Or are there other things happening inside of `InPlayer.Account.signOut`?

Reply:
this is only the case for simultaneous logins. When the simultaneous login limit is reached on the user login, we invalidate the user's token ( first session for the account ), and then, we send the notification to the user's device. Because the token is already invalidated, the request for logout fails( because of an invalid token ). I will try to find a better solution today.

@borkopetrovicc Maybe we can just update the state on logout notification?!

e.g.
```
         
case NotificationsTypes.ACCOUNT_LOGOUT:
await logout(true);
if (notification.resource?.reason === 'sessions_limit') {
   window.location.href = addQueryParams(window.location.href, { u: 'simultaneous-logins' });
}
break;
```

and extend the logout method from the controller

e.g.
```
export async function logout(onlyState = false) {
  await useService(async ({ accountService }) => {
    // this invalidates all entitlements caches which makes the useEntitlement hook to verify the entitlements.
    await queryClient.invalidateQueries('entitlements');

    useAccountStore.setState({
      user: null,
      subscription: null,
      transactions: null,
      activePayment: null,
      customerConsents: null,
      publisherConsents: null,
      loading: false,
    });

    await restoreFavorites();
    await restoreWatchHistory();
    if (onlyState) {
      await accountService?.logout();
    }
  });
}
```

Reply:
The case should be if (!onlyState) (with "!"), but other than that this seems like the best solution. Maybe better to call it sendNetworkRequest or something similar and have it be true by default

Reply:
The case should be if (!onlyState) (with "!"), but other than that this seems like the best solution. Maybe better to call it sendNetworkRequest or something similar and have it be true by default

Reply:
or even better, just move the logout part at the bottom

e.g.

```
case NotificationsTypes.ACCOUNT_LOGOUT:
if (notification.resource?.reason === 'sessions_limit') {
   window.location.href = addQueryParams(window.location.href, { u: 'simultaneous-logins' });
}
await logout();
break;
```

Reply:
Won't logout not be called because of using window.location.href in this case?

Reply:
correct, but on the page load, the user won't be authenticated because of the invalid session.

Reply:
I've tested and it is working ok. 95c9032c0c6cbe5a1e786fd357085f0108801488

Reply:
I think that with this function order, you can't guarantee that `await logout();` will be finished. This might not be a problem in the current situation, but I don't think relying on flaky behavior is a good idea (even though it still works).

The usage of `window.location.href` should still be discussed because it fully reloads the SPA, which isn't ideal for UX, PWA and Capacitor apps.

Reply:
45ec68a6df7215be2dfa5bfcbbaac6383f4aef50

Reply:
Can we put the logout into an else clause? Then it's more explicit that we are not intending to run logout when we refresh the page for simultaneous logout. A comment could also be nice.

Reply:
df8d70e2ad529888f73e9d1b9ca1a78db4d1a555